Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageClient.cs

```diff
@@ -158,24 +163,22 @@ public Task<TResult> InvokeWithParameterObjectAsync<TResult>(string targetName,
             => _rpc == null ? Task.FromResult(default(TResult)) : _rpc.InvokeWithParameterObjectAsync<TResult>(targetName, argument, cancellationToken);
 
         private void OnInterpreterChanged(object sender, EventArgs e) => SendDidChangeConfiguration().DoNotWait();
-        private void OnSearchPathsChanged(object sender, EventArgs e) => SendDidChangeConfiguration().DoNotWait();
+        private void OnAnalysisOptionsChanged(object sender, EventArgs e) => SendDidChangeConfiguration().DoNotWait();
 
         private async Task SendDidChangeConfiguration() {
             Debug.Assert(_clientContext != null);
-            // TODO: client context needs to also provide the settings that are currently hard coded here
-            // so that workspace can get it from PythonSettings.json and projects from their .pyproj, etc.
+            Debug.Assert(_analysisOptions != null);
+
             var settings = new Settings {
                 python = new Settings.PythonSettings {
                     pythonPath = _clientContext.InterpreterConfiguration.InterpreterPath,
                     venvPath = string.Empty,
                     analysis = new Settings.PythonSettings.PythonAnalysisSettings {
-                        logLevel = "log",
-                        autoSearchPaths = false,
-                        diagnosticMode = "openFilesOnly",
-                        diagnosticSeverityOverrides = new Dictionary<string, string>(),
-                        extraPaths = _clientContext.SearchPaths.ToArray(),
```

Comment:
We shouldn't remove support for `extraPaths`. What was there for it looked correct.

Reply:
Yes, I need to look into few more (and probably expose them), settings are scattered between Pylance, Pyright and core extension.

Reply:
Added more options